src/components/_global/BalForm/readme.md

```diff
@@ -0,0 +1,99 @@
+# BalForm
+
+`<BalForm />` is a useful component for creating wrappers for simple and complex form structures on our frontend. There are a few motivations behind using a form component for creating forms:
+
+- Centralised and instance specific state. The form owns its state and can pass down its state (aka form values) to children at any nested level. This also means not needing to creating and managing your own form state in whichever component / composable that might need to feed into your form.
+- Centralised validation. If your component accepts in a rules prop, you can integrate it as a form component (more on this below) and have it validate against all passed in rules, whenever the components value changes. Any errors present during submission of the form will prevent the submission function from executing and a list of errors for each field be exported by the forms instance. 
+- Simplified integration**. All you need to do is define the form component you are using as a child of `<BalForm />` to have its values, rules and any change handlers bound to the form, automatically.
+
+  ** Assuming that all child form components (e.g. inputs, radio buttons, complex inputs such as <TokenInput/> have code that allows them to integrate as a form component.
+
+
+## Usage
+
+```typescript
+// somecomponent.vue
+
+/// script
+
+// defines a component instance that will be
+// passed into <BalForm />. Names are tied to 
+// form instances and are unique. The value returned
+// represents the form instance.
+const form = useForm({ name: 'my-form' })
+
+function mySubmit(formValues: T) {
+ // do stuff
+}
+
+/// template
+<BalForm :form="form" :submit="form.handleSubmit(mySubmit)"
```

Comment:
This is quite an unintuitive usage, I'd expect to just pass in a function to the submit prop, like:
```suggestion
<BalForm :form="form" :submit="mySubmit"
```
Is there no way we can get it down to this?